vignettes/download_mail.Rmd

```diff
@@ -0,0 +1,215 @@
+
+The most manual time intensive step you will be required is to locate the URL of the mailing list archive you wish for in the project website. This is specified under `mailing_list`. Note for pipermail this URL should point to the page containing links to the monthly archives (e.g. https://mta.openssl.org/pipermail/openssl-users/), not the top-level mailing list page that contains all the different types of archives (e.g. https://mta.openssl.org/mailman/listinfo/).
+
+
+Regardless of which mail archive you choose, the downloaders will store the mail data in monthly files, in a `.mbox` format. This is a simple text file that contains some markings to identify the header of the e-mail containing title, authors, etc. While you can open any of the .mbox downloaded files with any text editor, Kaiaulu parsers will format them into tables, as demonstrated below. 
+
+## Tools Configuration
+
+In addition to the mailing list configurations, you need to specify the path to the [Perceval](https://github.com/chaoss/grimoirelab-perceval) binary in tools.yml. See the wiki for further details on how to setup third party tools. 
+
+```{r}
+# Load tools configuration
+tools <- parse_config("../tools.yml")
+parse_perceval_path <- get_tool_project("perceval", tools)
+
+# Load project configuration
+conf <- parse_config("../conf/helix.yml")
+mbox_file_path <- get_mbox_path(conf, "project_key_1")
+```
+
+# Downloaders and Refreshers
+
+## Downloaders
+
+With the configurations loaded, we can proceed to download the mailing list archives. The downloaders are responsible for fetching the archives from the specified mailing lists and saving them locally in .mbox format.
+
+### Pipermail Downloader
+
+For Pipermail, we need to specify the project key, which is used to retrieve the configuration parameters for the specific project. The project key is used to identify the project in the configuration file.
+
+Now, we can use the getter functions to retrieve the configuration parameters for the specified project key.
+
+```{r eval=FALSE}
+conf <- parse_config("../conf/openssl.yml")
+pipermail_mailing_list <- get_pipermail_domain(conf, "project_key_1")
+pipermail_save_folder_path <- get_pipermail_path(conf, "project_key_1")
+
+# Define the date range
+pipermail_start_year_month <- 202310
+pipermail_end_year_month <- 202405
+```
+
+With our configurations loaded, we can proceed to downloading the mailing list archives.
+
+```{r eval=FALSE}
+# Download archives
+download_pipermail(
+  mailing_list = pipermail_mailing_list,
+  start_year_month = pipermail_start_year_month,
+  end_year_month = pipermail_end_year_month,
+  save_folder_path = pipermail_save_folder_path,
+  verbose = TRUE
+)
+
+```
+
+After running this function, the .mbox files will be saved in the specified directory with filenames like 202310.mbox, 202311.mbox, etc, which can be parsed in a table:
+
+```{r}
+parsed_mail <- parse_mbox(
+  perceval_path = parse_perceval_path,
+  mbox_file_path = mbox_file_path
```

Comment:
I believe this is meant to parse the pipermail data which is stored in the pipermail folder path, however this is attempting to parse data from the mbox folder path: 
`mbox_file_path = mbox_file_path` should be changed to `mbox_file_path = pipermail_save_folder_path`.